libs/sdk-ui-kit/src/@ui/UiMenu/UiMenu.tsx

```diff
@@ -0,0 +1,187 @@
+    isDisabledFocusable = true,
+
+    ariaAttributes,
+}: UiMenuProps<InteractiveItemData, StaticItemData>): React.ReactNode {
+    const [controlType, setControlType] = React.useState<ControlType>("unknown");
+
+    const menuComponentRef = React.useRef<HTMLMenuElement>(null);
+
+    const isItemFocusable = React.useCallback(
+        (item?: IUiMenuItem<InteractiveItemData, StaticItemData>) => {
+            if (!item || item.type !== "interactive") {
+                return false;
+            }
+
+            return isDisabledFocusable || !item.isDisabled;
+        },
+        [isDisabledFocusable],
+    );
+
+    const [focusedId, setFocusedId_internal] = React.useState<string | undefined>(
+        () => getItemsInSubMenu(items, undefined)?.find(isItemFocusable)?.id,
+    );
+    const setFocusedId = React.useCallback<typeof setFocusedId_internal>((...args) => {
+        setFocusedId_internal(...args);
+        // Focus is lost when clicking on an item that opens a submenu. We need to refocus the menu.
+        menuComponentRef.current?.focus();
+    }, []);
+
+    const focusedItem = getInteractiveItem(items, focusedId);
+
+    const { itemRefs, setItemRef } = useItemRefs(focusedId);
+
+    const handleSelectItem = React.useCallback(
+        (item?: IUiMenuInteractiveItem<InteractiveItemData, StaticItemData>) => {
+            if (!item || item.isDisabled) {
+                return;
+            }
+
+            // If there is no subMenu, select the item
+            if (!item.subMenu) {
+                onSelect?.(item);
+                shouldCloseOnSelect && onClose?.();
+                return;
+            }
+
+            // Otherwise focus the first focusable child
+            const itemToFocus = getItemsInSubMenu(items, item.id)?.filter(isItemFocusable)[0];
+
+            if (!itemToFocus) {
+                return;
+            }
+
+            setFocusedId(itemToFocus.id);
+            menuComponentRef.current?.focus();
+        },
+        [isItemFocusable, items, onClose, onSelect, setFocusedId, shouldCloseOnSelect],
+    );
+
+    const handleFocusItemByMouse = React.useCallback(
+        (id: string) => () => {
+            if (controlType !== "mouse") {
+                return;
+            }
+            setFocusedId(id);
+        },
+        [controlType, setFocusedId],
+    );
+
+    const context = {
+        itemRefs,
+        focusedId,
+        items,
+        focusedItem,
+        onClose,
+        onSelect: handleSelectItem,
+        setFocusedId,
+        isItemFocusable,
+    };
+    const contextRef = useAutoupdateRef<IUiMenuContext<InteractiveItemData, StaticItemData>>(context);
+
+    const handleKeyDown = useKeyNavigation<InteractiveItemData, StaticItemData>({
+        menuContextRef: contextRef,
+        onUnhandledKeyDown,
+        shouldKeyboardActionPreventDefault,
+        shouldKeyboardActionStopPropagation,
+    });
+
+    const menuId = useId();
+
+    const subMenuItems = React.useMemo(
```

Comment:
currentMenuIterms would be better as in most cases it will be root menu items (submenus are not so common now)